kafka/oauth/abstract.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import absolute_import
+
+import abc
+
+# This statement is compatible with both Python 2.7 & 3+
+ABC = abc.ABCMeta('ABC', (object,), {'__slots__': ()})
+
+class AbstractTokenProvider(ABC):
+    """
+    A Token Provider must be used for the SASL OAuthBearer protocol.
+
+    The implementation shsould ensure token reuse so that multiple
+    calls at connect time do not create multiple tokens. The implementation
+    should also periodically refresh the token in order to guarantee
+    that each call returns an unexpired token. A timeout error should
+    be returned after a short period of inactivity so that the
+    broker can log debugging info and retry.
+
+    Token Providers MUST be implemented from this ABC.
+
+    An optional method that may be implemented if the user chooses is:
+    #extensions() - Returns a map of key-value pairs that can
```

Comment:
Consider including a stub method definition below. Don't decorate with `@abc.abstractmethod` so that it isn't required. But I think it adds more clarity to the class overall.